tempodb/compaction_block_selector.go

```diff
@@ -74,29 +78,82 @@ func newTimeWindowBlockSelector(blocklist []*backend.BlockMeta, maxCompactionRan
 		}
 
 		age := currWindow - w
+		builder.Reset()
+
 		if activeWindow <= w {
+			// Grow size: 2+20+1+16+1+20
+			builder.Grow(60)
```

Comment:
this is a very tiny %age of all compactor allocations and the previous implementation had a lot of clarity on what the group/order/hash strings looked like. the current implementation loses this a bit

i'm fine with moving forward with this, but can we need the new implementation to read as cleanly as the old given that the value gained is so small